crates/pipeline-manager/src/runner/pipeline_automata.rs

```diff
@@ -1066,9 +1066,53 @@ impl<T: PipelineExecutor> PipelineAutomaton<T> {
                 .expect("Provision must have been called"),
         );
         match self.pipeline_handle.is_provisioned().await {
-            Ok(Some(location)) => State::TransitionToInitializing {
-                deployment_location: location,
-            },
+            Ok(Some(location)) => {
+                match self
+                    .check_pipeline_status(self.pipeline_id, &location)
+                    .await
+                {
+                    StatusCheckResult::Unavailable
+                        if Self::has_timeout_expired(
+                            pipeline.deployment_status_since,
+                            provisioning_timeout,
+                        ) =>
+                    {
+                        error!(
+                            "Pipeline {} was provisioned, but didn't respond to status check in time",
```

Comment:
"Pipeline {} was provisioned, but didn't respond to status check in XXXX seconds - terminating the pipeline",